Sources/NextcloudFileProviderKit/Database/FilesDatabaseManager.swift

```diff
@@ -661,4 +661,20 @@ public final class FilesDatabaseManager: Sendable {
 
         return (updated, deleted)
     }
+    
+    public func itemsMetadataFromPattern(
+        pattern: String
+    ) -> [SendableItemMetadata]? {
+        logger.debug("Trying to find files matching pattern \(pattern)")
+        let results = itemMetadatas.where { $0.fileName.ends(with: pattern) && !$0.directory }
+        guard !results.isEmpty else {
+            logger.debug("Could not find files matching pattern \(pattern)")
+            return nil
```

Comment:
Related to my comment on the return type.
```suggestion
            return []
```